# Graded assignment 1

Pascale van de Ven 2013998

# Packages

We need the package `tidyverse`

In [11]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [12]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.9908371 0.5841975 0.8512326 0.4094907 0.1246844 0.4240438 0.7420997
2 0.7067653 0.6611356 0.8258662 0.3355499 0.8750663 0.5224260 0.1920198
  V8        V9        V10       V11       V12       V13       V14      
1 0.6101953 0.5985393 0.3360333 0.7884314 0.1688488 0.2377127 0.5301810
2 0.7890050 0.5988136 0.2098615 0.9882070 0.2994151 0.7641592 0.3914639
  V15         V16       V17        V18       V19       V20      
1 0.777419058 0.9115628 0.44700608 0.3677519 0.1996989 0.6208004
2 0.001508137 0.0192557 0.07075471 0.3605606 0.7020025 0.1901727

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [13]:
df2 <- df1 %>%
    gather (key, value)


head(df2)

key value    
1 V1  0.9908371
2 V1  0.7067653
3 V2  0.5841975
4 V2  0.6611356
5 V3  0.8512326
6 V3  0.8258662

## Assignment 2

We will generate some data:

In [14]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [10]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook